examples/vision/image_classification_efficientnet_fine_tuning.py

```diff
@@ -0,0 +1,423 @@
+"""
+
+from tensorflow.keras.applications import EfficientNetB0
+from tensorflow.keras.layers.experimental.preprocessing import (
+    Resizing,
+    RandomFlip,
+    RandomContrast,
+    # RandomHeight,
+)
+from tensorflow.keras.optimizers import SGD
+
+with strategy.scope():
+    inputs = keras.layers.Input(shape=(32, 32, 3))
+    x = inputs
+
+    x = RandomFlip()(x)
+    x = RandomContrast(0.1)(x)
+    # x = RandomHeight(0.1)(x)
+    x = Resizing(IMG_SIZE, IMG_SIZE, interpolation="bilinear")(x)
+
+    x = EfficientNetB0(include_top=True, weights=None, classes=100)(x)
+
+    model = keras.Model(inputs, x)
+
+    sgd = SGD(learning_rate=0.2, momentum=0.1, nesterov=True)
+    model.compile(optimizer=sgd, loss="categorical_crossentropy", metrics=["accuracy"])
+
+model.summary()
+reduce_lr = tf.keras.callbacks.ReduceLROnPlateau(
+    monitor="val_loss", factor=0.2, patience=5, min_lr=0.005, verbose=2
+)
+
+epochs = 20  # @param {type: "slider", min:5, max:50}
+hist = model.fit(
+    ds_train, epochs=epochs, validation_data=ds_test, callbacks=[reduce_lr], verbose=2
+)
+
+
+"""
+Training the model is relatively fast (takes only 20 seconds per epoch on TPUv2 that is
+available on colab). This might make it sounds easy to simply train EfficientNet on any
+dataset wanted from scratch. However, training EfficientNet on smaller datasets,
+especially those with lower resolution like CIFAR-100, faces the significant challenge of
+overfitting or getting trapped in local extrema.
+
+Hence traning from scratch requires very careful choice of hyperparameters and is
+difficult to find suitable regularization. Plotting the training and validation accuracy
+makes it clear that validation accuracy stagnates at very low value.
+"""
+
+import matplotlib.pyplot as plt
+
+
+def plot_hist(hist):
+    plt.plot(hist.history["accuracy"])
+    plt.plot(hist.history["val_accuracy"])
+    plt.title("model accuracy")
+    plt.ylabel("accuracy")
+    plt.xlabel("epoch")
+    plt.legend(["train", "validation"], loc="upper left")
+    plt.show()
+
+
+plot_hist(hist)
+
+"""
+### transfer learning from pretrained weight
+Using pre-trained imagenet weights and only transfer learn (fine-tune) the model allows
+utilizing the power of EfficientNet much easier. To use pretrained weight, the model can
+be initiated through
+"""
+
+from tensorflow import keras
+from tensorflow.keras.layers.experimental.preprocessing import (
+    Resizing,
+    RandomContrast,
+)
+
+
+def build_model(n_classes):
+    inputs = keras.layers.Input(shape=(32, 32, 3))
+    x = inputs
+
+    x = RandomFlip()(x)
+    x = RandomContrast(0.1)(x)
+    x = Resizing(IMG_SIZE, IMG_SIZE, interpolation="bilinear")(x)
+    # other preprocessing layers can be used similar to Resizing and RandomRotation
+
+    model = EfficientNetB0(include_top=False, input_tensor=x, weights="imagenet")
+
+    # freeze the pretrained weights
+    for l in model.layers:
```

Comment:
I think you can set model.trainable = False, which should set all the sub layer's trainable to False. Can u try that?

Reply:
Sure. As explicitly going through each layer is demonstrated when unfreezing the model, it is certainly good to use model.trainable